Jint/Native/Array/IArrayLike.cs

```diff
@@ -0,0 +1,7 @@
+﻿namespace Jint.Native.Array
+{
+    internal interface IArrayLike
```

Comment:
Looking at the spec, an "array-like" object is from TypedArray, and here only String implements it. 
https://www.ecma-international.org/ecma-262/6.0/#sec-typedarray-objects

Also in the Array.from you don't use Iterator at all, is that intentionally left for later?
https://www.ecma-international.org/ecma-262/6.0/#sec-array.from

Reply:
I see that what they call "array-like" is any object that has a "length" property and can have properties like `'0'`, ... So yes String is an array-like object. But maybe the interface should actually expose the `Length` and `this[int]` members? Might be easier to implement the spec that expects array-like objects.

Reply:
I started to check Array items and was overwhelmed with the stuff that is needed. So I was thinking maybe process this part first and when PR lands continue with adding missing bits to strings and Array.